tensorflow/python/ops/numpy_ops/np_array_ops.py

```diff
@@ -1027,16 +1027,19 @@ def broadcast_to(array, shape):  # pylint: disable=redefined-outer-name
 
 @np_utils.np_doc('stack')
 def stack(arrays, axis=0):  # pylint: disable=missing-function-docstring
-  if isinstance(arrays, (np_arrays.ndarray, ops.Tensor)):
-    arrays = asarray(arrays)
-    if axis == 0:
-      return arrays
-    else:
-      return swapaxes(arrays, 0, axis)
-  arrays = _promote_dtype(*arrays)  # pylint: disable=protected-access
-  unwrapped_arrays = [
+  if -(tf.rank(arrays)) <= axis < tf.rank(arrays):
```

Comment:
Please change `tf.rank` to `array_ops.rank`

Reply:
updated requested changes.Please review and confirm if anything missing.